02/implement-intersection.checkpoint.md

```diff
@@ -211,27 +211,187 @@ class TestChallenge(unittest.TestCase):
 ### !end-tests
 ### !explanation
 
-An example of a working implementation:
+Examples of Working Implementations:<br><br>
+
+An initial approach to this problem is to keep track of visited nodes. The solution would traverse the first linked list and store the addresses of the visited nodes in a set and then traverse the second linked list. While traversing the second list, if you encounter an address that already exists in the set then you've identified the intersecting node and can return it.<br><br>
 
 ```python
-# Time complexity: O(A + B) where A is the size of list A and B is the size of list B. The loop will traverse through the lists at most twice.
-# Space complexity: O(1) as the added space needed does not scale with the input data (two additional pointers)
 def intersection_node(head_a, head_b):
-    # assign l1 and l2 to point to the heads of list A and list B
-    l1, l2 = head_a, head_b
-    # while l1 and l2 do not reference the same node or None
-    while l1 != l2:
-        # once we've traversed through an entire list, we will set the pointer for the smaller list to the beginning of the other list
+    # create a set for storing the addresses of the nodes of the first linked list
+    nodes = set()
+
+    # traverse the first linked list with a loop and put each node's address in the set
+    while head_a:
+        nodes.add(id(head_a)) 
+        head_a = head_a.next
+    # then loop through second linked list and check if each node's address is in the set
+    while head_b:
+        if id(head_b) in nodes: 
+            # if node is in the set, then return that node 
+            return head_b
+        head_b = head_b.next
+    
+    # if there is no intersection, return None
+    return None
+```
+
+***
+
+**Note About id():**<br>
+In the solution above, we used the Python id function to get the address of each node object. However, in Python, user classes (or developer-created classes) are implicitly hashable which means Python takes their id automatically. It is not necessary or encouraged to use the id function in general code, but is done so here for clarity. Therefore, instead of `nodes.add(id(head_a))` we can just write `nodes.add(head_a)` and instead of `if id(head_b) in nodes` and we can just write `if head_b in nodes` and get the same result.<br><br>
```

Comment:
Code ticks when refering to the id fucntion.
```suggestion
In the solution above, we used the Python `id` function to get the address of each node object. However, in Python, user classes (or developer-created classes) are implicitly hashable which means Python takes their id automatically. It is not necessary or encouraged to use the `id` function in general code, but is done so here for clarity. Therefore, instead of `nodes.add(id(head_a))` we can just write `nodes.add(head_a)` and instead of `if id(head_b) in nodes` and we can just write `if head_b in nodes` and get the same result.<br><br>
```